packages/metadata/src/Metadata/util/validateTypes.ts

```diff
@@ -20,7 +20,7 @@ function extractTypes (types: string[]): any[] {
       case TypeDefInfo.Compact:
       case TypeDefInfo.Option:
       case TypeDefInfo.Vec:
-        return extractTypes([(decoded.sub as TypeDef).type]);
+        return extractTypes([(decoded.sub as TypeDef[])[0].type]);
```

Comment:
Can just combine this with the ones below, i.e. like in Tuple just iterate through the array, less moving parts and logic.